teps/0154-concise-remote-resolver-syntax.md

```diff
@@ -0,0 +1,326 @@
+
+To provide a consistent format across resolvers, we suggest the following `remote-resolver` template: `<scheme>://<location>[@<version>][#<selector>][?<params>]`. This template is a "best practice" so that all resolvers follow a similar template. IT will not be enforced by the Tekton Pipeline/Task controllers.
+
+Where, `scheme` is required while `location`, `version`, `selector` and `params` are all optional and should be parsed by the `Remote Resolvers` according to their needs. Additionally, the value of the `name` must be `url encoded`.
+In [stage 2](#stage-2), remote resolvers will subscribe to `scheme`s that they support and Tekton will auto-forward the resolution request to the subscribed resovers. 
+
+The `Remote Resolver` must define the format using the above `remote-resolver` template. The `Remote Resolver` is be responsible for parsing the `resolver-reference` and extracting the necessary information. The Tekton controller will perform no validation on the value of the `name`. It will simply pass it as is in the `Remote Resolution CRD`. 
+
+The `Remote Resolver` does not need to be able to parse all possible schemes. It can document the schemes it understands and what the underlying format it uses to understand the resolver reference. For instance, the git resolver might only understand schems `https`, `git+https`, `git`, `scmType` have the underlying resolver reference format suited to it.
+
+Examples:
+```
+https://github.com/tektoncd/test@main#task.yaml
+git+https://github.com/tektoncd/test@main#task.yaml
+git://github.com/tektoncd/test@main#task.yaml
+github://tektoncd/test@main#task.yaml
+oci://tektoncd/image:v1@sha256:...#task/foo?secret=my-secret
+https://example.com/task.yaml
+http://10.5.3.2:8080/task.yaml
+artifact/tasks-catalog/tasks/task@v1.0 # e.g. for the hub resolver where there is no need for a scheme
+```
+
+##### Git Resolver
+Based on `git` resolver's [params](https://tekton.dev/docs/pipelines/git-resolver/#parameters), we define the following `resolver-reference` formats:
+
+If scheme is `git`, `git+https` or `https`,
+
+```
+<git/git+https/https>://<complete git repo>@<revision>#<pathInRepo>?token=<token>&tokenKey=<tokenKey>
+```
+
+Example:
+
+```
+https://github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token
+git://github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token
+git+https://github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token
+```
+
+Using the above example, a user can do the following when referencing the Task:
+```yaml
+taskRef:
+  name: https://github.com/tektoncd/catalog@sha256:053a6cb9f3711d4527dd0d37ac610e8727ec0288a898d5dfbd79b25bcaa29828#task/hello-world?token=git-token
+  resolver: git
+```
+
+Alternatively, if the `scheme` is the `scmType` then the format is:
```

Comment:
That would be tricky because it would be treated as a different `scheme`. May be thats a limitation of this scheme. 
One way I see passing this information is via the previous scheme:
`https://github.my.com/...`

Reply:
Or, in such scenarios, the user can provide the exact resolver they wish to forward the request to so that Tekton does not need to figure out which ones to send the request to.

Reply:
I've reflected this point in the TEP.